src/distributions/uniform.rs

```diff
@@ -534,59 +533,110 @@ macro_rules! uniform_float_impl {
 
             fn new(low: Self::X, high: Self::X) -> Self {
                 assert!(low < high, "Uniform::new called with `low >= high`");
-                let scale = high - low;
-                let offset = low - scale;
-                UniformFloat {
-                    scale: scale,
-                    offset: offset,
+                assert!(low.is_finite() && high.is_finite(), "Uniform::new called with non-finite boundaries");
+
+                let mut scale = high - low;
```

Comment:
Can overflow to infinite even if both inputs are finite. Skip the above check (you already know neither value is an NaN) and just assert `scale < INFINITY`.

Also means you can simplify your `Float` trait.

Reply:
I intentionally didn't want to panic when `high` and `low` are both finite numbers. Even if `high-low` results in a non-finite number. Mainly because it felt like a lot to ask for users to know when that difference might overflow into infinity. But a little because depending on that the difference can be expressed by an $ty is technically an implementation detail. Another implementation of this function might not.

Either way, the code below deals fine with if `scale` becomes infinite here.